modules/ingester/instance_search.go

```diff
@@ -427,17 +450,59 @@ func (i *instance) SearchTagValuesV2(ctx context.Context, req *tempopb.SearchTag
 			return nil
 		}
 
-		// if the query is empty, use the old search
-		if traceql.IsEmptyQuery(query) {
-			return s.SearchTagValuesV2(ctx, tag, traceql.MakeCollectTagValueFunc(valueCollector.Collect), common.DefaultSearchOptions())
+		// cacher not provided, just search the block
+		if bc == nil {
```

Comment:
Since only a few lines of code of this func are used when not caching - what do you think about splitting it up into caching and non-caching versions?

Reply:
done